pkg/querier/batch/batch_test.go

```diff
@@ -55,6 +56,22 @@ func BenchmarkNewChunkMergeIterator_CreateAndIterate(b *testing.B) {
 	}
 }
 
+func TestSeekCorrectlyDealWithSinglePointChunks(t *testing.T) {
+	chunkOne := mkChunk(t, model.Time(1*step/time.Millisecond), 1, promchunk.PrometheusXorChunk)
+	chunkTwo := mkChunk(t, model.Time(10*step/time.Millisecond), 1, promchunk.PrometheusXorChunk)
+	chunks := []chunk.Chunk{chunkOne, chunkTwo}
+
+	sut := NewChunkMergeIterator(chunks, 0, 0)
+
+	// Following calls mimics Prometheus's query engine behaviour for VectorSelector.
+	sut.Next()
```

Comment:
It would be good to verify return values of these calls too.

Reply:
good idea